examples/nas/search_space_zoo/nas_bench_201.py

```diff
@@ -0,0 +1,167 @@
+        layer_reductions = [False] * N + [True] + [False] * N + [True] + [False] * N
+
+        C_prev = C
+        self.cells = nn.ModuleList()
+        for i, (C_curr, reduction) in enumerate(zip(layer_channels, layer_reductions)):
+            if reduction:
+                cell = ResNetBasicBlock(C_prev, C_curr, 2, self.bn_affine, self.bn_momentum, self.bn_track_running_stats)
+            else:
+                cell = NASBench201Cell(i, C_prev, C_curr, 1, self.bn_affine, self.bn_momentum, self.bn_track_running_stats)
+            self.cells.append(cell)
+            C_prev = C_curr
+
+        self.lastact = nn.Sequential(
+            nn.BatchNorm2d(C_prev, momentum=self.bn_momentum),
+            nn.ReLU(inplace=True)
+        )
+        self.global_pooling = nn.AdaptiveAvgPool2d(1)
+        self.classifier = nn.Linear(C_prev, self.num_labels)
+
+    def forward(self, inputs):
+        feature = self.stem(inputs)
+        for cell in self.cells:
+            feature = cell(feature)
+
+        out = self.lastact(feature)
+        out = self.global_pooling(out)
+        out = out.view(out.size(0), -1)
+        logits = self.classifier(out)
+
+        return logits
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser("nb201")
+    parser.add_argument('--stem_out_channels', default=16, type=int)
+    parser.add_argument('--unrolled', default=False, action='store_true')
+    parser.add_argument('--batch-size', default=64, type=int)
+    parser.add_argument('--epochs', default=50, type=int)
+    parser.add_argument('--num_modules_per_stack', default=5, type=int)
+    parser.add_argument('--log-frequency', default=10, type=int)
+    parser.add_argument('--bn_momentum', default=0.1, type=int)
+    parser.add_argument('--bn_affine', default=True, type=bool)
+    parser.add_argument('--bn_track_running_stats', default=True, type=bool)
+    parser.add_argument('--arch', default=None, help='json file which should meet requirements in NAS-Bench-201')
+    parser.add_argument('--visualization', default=False, action='store_true')
+    args = parser.parse_args()
+
+    dataset_train, dataset_valid = datasets.get_dataset("cifar10")
+    model = NASBench201Network(stem_out_channels=args.stem_out_channels,
+                               num_modules_per_stack=args.num_modules_per_stack,
+                               bn_affine=args.bn_affine,
+                               bn_momentum=args.bn_momentum,
+                               bn_track_running_stats=args.bn_track_running_stats)
+
+    optim = torch.optim.SGD(model.parameters(), 0.025)
+    lr_scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optim, args.epochs, eta_min=0.001)
+    criterion = nn.CrossEntropyLoss()
+
+    if args.arch is not None: 
+        with open(args.arch, 'r') as f:
+            arch = json.load(f)
+        for trial in query_nb201_trial_stats(arch, 200, 'cifar100'):
+            pprint.pprint(trial)
+        apply_fixed_architecture(model, args.arch)
```

Comment:
After applied fixed architecture, you should directly train it instead of using EnasTrainer.